cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -625,6 +625,17 @@
     </description>
   </property>
 
+  <property>
+    <name>plugin.exclude.with.requirements</name>
```

Comment:
Why use exclusion rather that listing what are supported in the instance (inclusion)?

Reply:
The exclusion approach looks clearer for the following reasons: 
1. If we take inclusion approach then we depend on the admin to specify all the possible requirements which are meet. So the default value has to include all the requirements meet.
2. The inclusion can be large list and hence someone will need to know them all and specify whereas exclusion will be a small list and hence easier to know and specify. If nothing is specified clearer to assume nothing is excluded.
3. With exclusion missing to specify a requirement will lead to plugin being displayed and then fail while running which can then be identified and added to exclusion list whereas for inclusion missing a requirement will lead to plugin not showing up at all and can lead to some confusion.

Reply:
1. Shouldn't the admin should knows what are the supported in a given environment before making changes?
2. How large the list possibly be? We are talking about system abilities here, right?
3. What are the confusions?

Reply:
It can be few requirements lets say 4-6 but we are relying on them to be specified to start with whereas we can easily work without them. Confusion of plugins not showing up. For example, if an admin in an instance does not say spark1 and/or spark2 and a plugin which is only compatible with spark1 is deployed will not be visible leading to the confusion of what happened by the the person who is deploying the jar. 

What is the issue with having it as exclusion? Are there any demerits which favor inclusion more?

Reply:
Had in person discussion and have decided to keep it exclusion for now and change the names. 
This commit has the changes discussed in person: https://github.com/caskdata/cdap/pull/10550/commits/e277b22c4993eadadcbdc1ab6bed61287642bb72